pyomo/contrib/doe/Tests/test_reactor_example.py

```diff
@@ -0,0 +1,155 @@
+# (1) Redistributions of source code must retain the above copyright notice, this list of conditions and the
+# following disclaimer.
+# (2) Redistributions in binary form must reproduce the above copyright notice, this list of conditions and
+# the following disclaimer in the documentation and/or other materials provided with the distribution.
+# (3) Neither the name of the Carbon Capture Simulation for Industry Impact, TRIAD, LLNS, BERKELEY LAB,
+# PNNL, UT-Battelle, LLC, ORNL, NOTRE DAME, PITT, UT Austin, TOLEDO, WVU, U.S. Dept. of Energy nor
+# the names of its contributors may be used to endorse or promote products derived from this software
+# without specific prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+# THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# 
+# You are under no obligation whatsoever to provide any bug fixes, patches, or upgrades to the features,
+# functionality or performance of the source code ("Enhancements") to anyone; however, if you choose to
+# make your Enhancements available either publicly, or directly to Lawrence Berkeley National Laboratory,
+# without imposing a separate written license agreement for such Enhancements, then you hereby grant
+# the following license: a non-exclusive, royalty-free perpetual license to install, use, modify, prepare
+# derivative works, incorporate into other computer software, distribute, and sublicense such
+# enhancements or derivative works thereof, in binary and source code form.
+#
+# Lead Developers: Jialu Wang and Alexander Dowling, University of Notre Dame
+#
+#################################################################################################################
+
+# import libraries
+from pyomo.common.dependencies import (
+    numpy as np, numpy_available,
+    pandas as pd, pandas_available,
+)
+
+import pyomo.common.unittest as unittest
+from pyomo.contrib.doe.fim_doe import Measurements, DesignOfExperiments
+
+from pyomo.opt import SolverFactory
+ipopt_available = SolverFactory('ipopt').available()
+
+class doe_object_Tester(unittest.TestCase):
+    """ Test the kinetics example with both the sequential_finite mode and the direct_kaug mode
+    """
+    @unittest.skipIf(not ipopt_available, "The 'ipopt' command is not available")
+    def setUP(self):
```

Comment:
This isn't actually running a test of the reactor kinetics example. Test methods must have names starting with `test` to be picked up by pytest. I'd recommend restructuring this test similar to the parmest example tests. See: https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/parmest/tests/test_examples.py

In particular, I suggest creating a `main` function in `reactor_kinetics.py` that includes all the Pyomo.DOE calls so that you have a complete user example. I know you're including the Jupyter notebook as a complete example but for users who prefer to work outside of Jupyter I think it's still valuable to provide a complete example script.